rfcs/2021-11-22-design-tokens.md

```diff
@@ -0,0 +1,106 @@
+# RFC: Implementation of design tokens
+
+**Date** : Nov 22, 2021
+**Status** : Review
+
+## Problem
+
+Presently, the inconsistency of design rules between the DSM, Designers making and consuming changes, and the design system codebase lead to **confusion**, **duplicated work**, **inconsistency** and **disorganization** within all three systems.
+
+## Proposal
+
+[Design tokens](https://www.invisionapp.com/inside-design/design-tokens/) can help us solve this problem.
+
+> Design tokens are all the values needed to construct and maintain a design system — spacing, color, typography, object styles, animation, etc. — represented as data. These can represent anything defined by design: a color as a RGB value, an opacity as a number, an animation ease as Bezier coordinates. They’re used in place of hard-coded values in order to ensure flexibility and unity across all product experiences. - [Adobe Spectrum](https://spectrum.adobe.com/page/design-tokens/)
+
+Design tokens can be packaged from the atomic level to the application level by utilizing them in a dependency tree where one value inherits a value from the trunk above it.
+
+A design token solution is needed in which:
+
+1. The set of design tokens remain finite and equal
+2. The set of design tokens maintain consistency between application (DSM, Sketch, React and other future platforms such as iOS, Android, Web Components)
+3. The set of design tokens can be transposed, updated and extended easily and quickly across all systems at once
+4. The set of design tokens can be versioned and changes in that system can be tracked and monitored
+
+By deciding on a [Single Source Of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) (SSOT) for our design rules, we can organize and transform them for any system which utilizes them.
+
+## Benefits
+
+1. **Consistency** - Between design and code, utilizing a single set of variables everywhere allows a design proof to accurately describe the visual representation of a component as a set of variables. This provides better parity between what's designed and what is seen in the rendered usage on all platforms.
+
+   Looking at the [CSSStats for healthcare.gov/see-plans](https://cssstats.com/stats?url=healthcare.gov/see-plans), there are currently 29 color variations, many of which are nearly identical. Utilizing design tokens could bring this number down considerably and provide better brand consistency.
+
+2. **Organization** - By starting from the atomic level, we can use naming conventions for our tokens that are simple and specific to the variable they represent. By keeping these tokens separate from the systems they are utilized in, they can be managed as a discrete unit. Agreement on naming conventions of variables could eventually create parity between Sketch libraries, DS Sass variables, and future design and development tooling.
```

Comment:
I'm really curious to get into how code and Sketch naming can and should be in lock step in terms of naming

Reply:
Our conversation Monday should help but also you can impromptu hit me up anytime to chat unless I'm marked as away/busy in slack.